dapr_agents/agents/durable.py

```diff
@@ -548,8 +548,27 @@ async def _execute_tool() -> Any:
                 if hasattr(entry, "last_message"):
                     entry.last_message = tool_message_model
 
-        # Always persist to memory + in-process tool history
-        self.memory.add_message(tool_message)
+        # Check if tool message already exists in persistent memory before adding
+        tool_call_id = agent_message["tool_call_id"]
+        existing_memory_messages = self.memory.get_messages()
+        tool_exists_in_memory = False
+        for mem_msg in existing_memory_messages:
+            msg_dict = (
```

Comment:
Is this necessary?

```python
            msg_dict = (
                mem_msg.model_dump()
                if hasattr(mem_msg, "model_dump")
                else (mem_msg if isinstance(mem_msg, dict) else {})
            )
   ```
   
   The get_message methods always return a list of dictionaries right?